superset/security/manager.py

```diff
@@ -1337,13 +1341,22 @@ def get_guest_user_from_request(self, req: Request) -> Optional[GuestUser]:
             return None
 
         try:
+            aud = (
+                current_app.config["GUEST_TOKEN_JWT_DECODE_AUDIENCE"] or get_url_host()
+            )
             token = self.parse_jwt_guest_token(raw_token)
```

Comment:
FWIW, `aud` validation is provided by the `jwt.decode()` method See: https://github.com/jpadilla/pyjwt/blob/master/jwt/api_jwt.py#L147 Suggest using this :)

Reply:
https://github.com/jpadilla/pyjwt/blob/master/jwt/api_jwt.py#L32 `verify_aud` is set to True by default